lib/auth/scram.js

```
@@ -296,7 +294,10 @@ ScramSHA1.prototype.auth = function(server, connections, db, username, password,
 
   var _execute = function(_connection) {
     process.nextTick(function() {
-      executeScram(_connection);
+      crypto.randomBytes(24, function(eIgnore, buff) {
+        var nonce = buff.toString('base64');
```

can you please throw an error here if it exists?